src/lib.rs

```diff
@@ -1,24 +1,61 @@
 //! # Introduction
-//! Clarity is a low level library designed to handle Ethereum transactions.
-//!
-//! You can create, sign, verify transactions, as well as work with contract calls, and encode ABI.
-//! It is designed with embedded devices on mind so big/little endian architectures and 32/64 bits are supported.
+//! Clarity is a low-level Ethereum transaction library written in pure Rust.
 //!
 //! ## Features
+//! * Any-endian, 32/64-bit support
+//! * Public/private key handling
+//! * Transaction signing and verification
+//! * ABI enconding for common data types (see `abi::Token` variants)
+//!
+//! ## Getting started
+//! Here's an example lifetime of an Alice-to-Bob Ethereum transaction made with Clarity:
+//! ```rust
+//! extern crate clarity;
+//! extern crate futures;
+//! extern crate web3;
+//!
+//! use clarity::{Address, Signature, Transaction, PrivateKey};
+//!
+//! use futures::Future;
+//! use web3::{transports, types::Bytes, Web3};
+//!
+//! // A helper for filling the keys
+//! let mut key_buf: [u8; 32] = rand::random();
+//!
+//! let alices_key = PrivateKey::from_slice(&key_buf).unwrap();
+//!
+//! key_buf = rand::random();
+//! let bobs_key = PrivateKey::from_slice(&key_buf).unwrap();
+//!
+//! // Create a new transaction
+//! let tx = Transaction {
+//!     nonce: 0.into(),
+//!     gas_price: 1_000_000_000.into(),
+//!     gas_limit: 21_000.into(),
+//!     to: bobs_key.to_public_key().unwrap(),
+//!     value: 100.into(),
+//!     data: Vec::new(),
+//!     signature: None, // Not signed. Yet.
+//! };
 //!
-//! * Supports both little endian and big endian
-//! * Works well on both 32 and 64 bit architectures
-//! * Handle private keys
-//! * Create public keys
-//! * Sign transactions
-//! * Verify transaction
-//! * Handle signatures
-//! * Encode ABI for contract calls
+//! let tx_signed: Transaction = tx.sign(&alices_key, None);
+//! assert!(tx_signed.is_valid());
 //!
-//! ## Documentation
+//! // You can always derive the sender from a signed transaction
+//! let sender: Address = tx_signed.sender().unwrap();
 //!
-//! * [GitHub repository](https://github.com/althea-mesh/clarity)
-//! * [Cargo package](https://crates.io/crates/clarity)
+//! /*
```

Comment:
I just saw the render locally. What about changing it to a different section that would hide everything other than sign code? It would be mostly copy paste but you can hide all lines you don't want to show again with '#' prefix in the line